libs/model-ad/model-details/src/lib/model-details.component.ts

```diff
@@ -1,9 +1,134 @@
-import { Component } from '@angular/core';
+import { CommonModule, Location } from '@angular/common';
+import { AfterViewInit, Component, DestroyRef, inject, OnInit } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
+import { Panel } from '@sagebionetworks/explorers/models';
+import { HelperService } from '@sagebionetworks/explorers/services';
+import { PanelNavigationComponent } from '@sagebionetworks/explorers/ui';
+import { LoadingIconComponent } from '@sagebionetworks/explorers/util';
+import { Model, ModelsService } from '@sagebionetworks/model-ad/api-client-angular';
+import { ConfigService } from '@sagebionetworks/model-ad/config';
 
 @Component({
   selector: 'model-ad-model-details',
-  imports: [],
+  imports: [CommonModule, PanelNavigationComponent, LoadingIconComponent],
   templateUrl: './model-details.component.html',
   styleUrls: ['./model-details.component.scss'],
 })
-export class ModelDetailsComponent {}
+export class ModelDetailsComponent implements OnInit, AfterViewInit {
+  route = inject(ActivatedRoute);
+  router = inject(Router);
+  location = inject(Location);
+  helperService = inject(HelperService);
+  modelsService = inject(ModelsService);
+  destroyRef = inject(DestroyRef);
+  configService = inject(ConfigService);
+
+  isLoading = true;
+
+  model: Model | undefined;
+
+  panels: Panel[] = [
+    {
+      name: 'omics',
+      label: 'Omics',
+      disabled: false,
+    },
+    {
+      name: 'biomarkers',
+      label: 'Biomarkers',
+      disabled: false,
+    },
+    {
+      name: 'pathology',
+      label: 'Pathology',
+      disabled: false,
+    },
+    {
+      name: 'resources',
+      label: 'Resources',
+      disabled: false,
+    },
+  ];
+
+  activePanel = 'omics';
+  activeParent = '';
+
+  ngOnInit() {
+    this.route.paramMap.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((params: ParamMap) => {
+      this.isLoading = true;
+      if (this.configService.config.isPlatformServer) {
```

Comment:
as an enhancement for readability, you could wrap the rest of the logic in a method so the intent is clearer.  For example (pseudocode):
```
ngOnInit() {
  // only fetch data during client hydration
  if (isPlatformBrowser(this.platformId)) {
     this.loadPanels();
  }
}

loadPanels() {
  (rest of logic here)
}
```